internal/driver/driver.go

```diff
@@ -199,6 +271,7 @@ func (d *driver) getStorageByName(storageName string) (*upcloud.Storage, error)
 	var found bool
 	var storage upcloud.Storage
 	for _, s := range response.Storages {
+		// TODO: should we compare are these strings equal instead ?
```

Comment:
Seems to me as well that this might result in false matches

Reply:
I will address these TODO comments in separate PR as those affect also the builder